compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -720,7 +720,8 @@ impl<'a, 'b, 'tcx> TypeVerifier<'a, 'b, 'tcx> {
                 }
                 PlaceTy::from_ty(fty)
             }
-            ProjectionElem::OpaqueCast(ty) => {
+
+            ProjectionElem::Subtype(ty) | ProjectionElem::OpaqueCast(ty) => {
```

Comment:
actually, `Subtype` should simply be unreachable during borrowck, shouldn't it

Reply:
Yes, we run `Subtyper` after borrowck